File: tests/exercises/lotto/solution/correct.kt
import java.util.concurrent.ThreadLocalRandom

fun loterij(aantal: Int = 6, maximum: Int = 42): String {
    val randomGenerator = ThreadLocalRandom.current()
    val result: HashSet<Int> = HashSet()
    while (result.size < aantal) {
        result += randomGenerator.nextInt(1, maximum + 1)
    }
    return result.asSequence()
            .sorted()
            .joinToString(
                    separator = " - ",
                    transform = { i -> i.toString() }
            )
}

File: tests/exercises/lotto/solution/correct.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Submission {
    public static string Loterij(int aantal = 6, int maximum = 42) {
        var r = new Random();
        var result = new HashSet<int>();
        while (result.Count < aantal) {
            result.Add(r.Next(1, maximum + 1));
        }
        var list = result.ToList()
                        .OrderBy(i => i)
                        .Select(i => i.ToString());
        return String.Join(" - ", list);
    }
}
